### PR TITLE
Add Slim add-on

### PR DESCRIPTION
**Pull request by rafayet-monon**
## What happened
Solved #213 

At Nimble we use Slim for most of the project. In this PR we added Slim as an add-on to minimize the project setup time.
 
## Insight

- [x] Add Slim as add-on
- [x] Add slim-lint to danger system

We added the `slim` gem to Gemfile and `danger-slim_lint` danger plugin to lint slim files.
 

## Proof Of Work

- Ask while creating the project - 

image

- Associated gems added by creating from this branch -

image

image

 - add `slim_lint` `Dangerfile` after creating a project from this branch

image

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -11,14 +13,15 @@ run:
...
+slim_spec = spec/addons/slim/**/*_spec.rb
 test:
 	cd $(APP_NAME) && \
 	docker-compose -f docker-compose.test.yml up --detach db redis && \
 	docker-compose -f docker-compose.test.yml run --detach test bin/start.sh && \
 	cd ../.template && \
 	bundle install; \
 	if [ $(VARIANT) = web ]; then \
-	  bundle exec rspec --pattern="$(base_spec),spec/variants/web/**/*_spec.rb"; \
+	  bundle exec rspec --pattern="$(base_spec),spec/variants/web/**/*_spec.rb,$(slim_spec)"; \
```
Comment (andyduong1920) [inline Makefile:24]: The Makefile was refactored in #248, so please rebase this PR @rafayet-monon 🙏

File: Makefile [hunk 2]
```diff
@@ -1,5 +1,7 @@
+# Y - in response to Would you like to add Slim Template Engine(yes/no)
+
 create:
-	rails new $(APP_NAME) -m ./template.rb -T ${OPTIONS}
+	printf "Y\n" | rails new $(APP_NAME) -m ./template.rb -T ${OPTIONS}
```
Comment (andyduong1920) [inline Makefile:4]: Same with this [comment](https://github.com/nimblehq/rails-templates/pull/239#discussion_r501496294)

File: .semaphore/semaphore.yml [hunk 3]
```diff
@@ -79,7 +79,7 @@ blocks:
         - name: Create
           commands:
             - gem install rails -v $RAILS_VERSION --no-document
-            - make create OPTIONS="--api"
+            - make create_api OPTIONS="--api"
```
Comment (andyduong1920) [inline .semaphore/semaphore.yml:82]: since we have `create_api` already, should we moving the `--api` to Makefile too 🤔

**Comment by andyduong1920** [comment]
@rafayet-monon let update the available addon on README too

Pasted_Image_10_9_20__10_20

File: .semaphore/semaphore.yml (hunk 3 above)
Comment (rafayet-monon) [reply .semaphore/semaphore.yml]: Added in [16f5f5f](https://github.com/nimblehq/rails-templates/pull/237/commits/16f5f5fc5a6332819f74436dd7719b53c97e5d97)

**Review by andyduong1920 — APPROVED** [review]
LGTM @rafayet-monon

**Review by noor-ahmed — APPROVED** [review]
LGTM 💯

**Review by gutakk — APPROVED** [review]
Looks good to me

File: Makefile [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # Y - in response to Would you like to add PhraseApp configuration?
```
Comment (andyduong1920) [inline Makefile:1]: @rafayet-monon could you help me to adjust this message too 🙏  in response to Would you like to add **the** PhraseApp configuration?

Comment (rafayet-monon) [reply Makefile]: Fixed in [3214e7e](https://github.com/nimblehq/rails-templates/pull/237/commits/3214e7ef3db81e6f10b8207155aa6d00ee169ef3)

**Pull request merged**